Automattic/newspack-plugin
# Fix Site Kit Handoffs

### All Submissions:






### Changes proposed in this Pull Request:

This PR fixes a few issues with handoffs to Site Kit:

- Analytics Wizard: Clicking "View" on the Google Analytics card will activate the Site Kit Analytics module before performing the handoff. Previously, if Analytics  wasn't active this would lead to a access forbidden message.
- Advertising Wizard: The AdSense toggle is now directly tied to the activation state of the Site Kit AdSense module. Previously this toggle reflected a Newspack option, meaning it was not necessarily an accurate representation of Site Kit.
- Advertising Wizard: Previously there was an AdSense screen, which was only a handoff button. This has been removed and now the Configure link on  the  AdSense Card leads directly to the Site Kit AdSense module page.
- Advertising Wizard: Clicking the Configure link on the AdSense Card will activate the module before performing the handoff.

### How to test the changes in this Pull Request:

1. Navigate to Site Kit and disable Analytics and AdSense modules. This is  done by going to Settings, clicking the downward arrow by the Service (assuming it is active), clicking Edit, then "Disconnect {SERVICE} from Site Kit" 
2. Navigate to Newspack->Analytics. Click "View" next to Google Analytics. Verify you are taken to a valid page for Site Kit Analytics, and that it is active.
3. Navigate to Newspack->Advertising. Verify Google AdSense toggle  is off.
4. Switch on Google AdSense toggle, then click Configure.
5. Verify you are taken directly to the Site Kit AdSense page, and that the interim screen has been removed. 
6. Navigate back to Newspack->Advertising and verify Google AdSense toggle is on.

### Other information:
